lib/wallet.py

```diff
@@ -1437,6 +1438,13 @@ def receive_history_callback(self, addr, hist, tx_fees):
         if self.network:
             self.network.trigger_callback('on_history', self)
 
+    def add_tx_to_history(self, txid):
+        with self.lock:
+            for addr in itertools.chain(list(self.txi.get(txid, {}).keys()), list(self.txo.get(txid, {}).keys())):
+                cur_hist = self._history.get(addr, list())
+                cur_hist.append((txid, 0))
```

Comment:
Shouldn't you be checking for dupes here? Like not add the same txid multiple times...?

Reply:
Probably more efficient would be if _history would contain set's, but that's probably outside the scope of this PR. Will add if check

Reply:
Yeah it will be slow I agree.. *sigh*. It is what it is.  It's better to add the slow check in order to maintain wallet sanity and take the hit to performance here because.. this is a hack anyway. Might as well complete the hack!

Yeah..  It doesn't contain sets or a dict i presume because it needs to maintain some sort of order... *shrug*

Reply:
Fixed. But when backporting from electrum I saw they are using sets. Maybe they sort when outputting.